src/github.com/mozilla-services/pushgo/simplepush/error.go

```diff
@@ -63,8 +66,8 @@ func ErrToStatus(err error) (status int, message string) {
 	if err == nil {
 		return http.StatusOK, ""
 	}
-	if code, ok := err.(ErrorCode); ok {
-		switch status = code.Status(); status {
+	if se, ok := err.(*ServiceError); ok {
+		switch status = se.Status(); status {
```

Comment:
Backward-compatibility, mostly. :wink: 1.4 and earlier only return 401, 500, and 503 responses, and generic error strings.

That said, both clients only check the status code of the register response—Simple Push treats [non-200 codes as errors](https://github.com/mozilla/gecko-dev/blob/master/dom/push/PushService.jsm#L1031-1035), and Loop Push [retries on 500](https://github.com/mozilla/gecko-dev/blob/master/browser/components/loop/MozLoopPushHandler.jsm#L724-729). I'd be okay with surfacing the "real" error messages and status codes. What do you think, @jrconlin?